### PR TITLE
Ignore last committer instead of PR opener

### LINKED ISSUE #284: Change "no self-approval" logic
Currently we just ignore the PR opener to prevent people from self-approving pull requests. However it would be better to ignore the last committer instead.

Consider the following case: 

1. Alice commits stuff, opens pull request 1
2. Bob commits stuff on PR 1 too
3. Bob can now legally approve his own changes.

### PR DESCRIPTION
**Pull request by prayerslayer**
Fixes #284 

And sorry for the indentation noise, this is IntelliJ's fault.

### CODE REVIEW

File: server/checks/Approval.js [hunk 1]
```diff
@@ -139,9 +138,12 @@ export default class Approval extends Check {
             return
           }
           // get approvals for pr
-          const opener = pull_request.user.login
+          const commits = await github.fetchPullRequestCommits(user, repo, number, token)
+          const lastCommitter = commits.length === 0 ?
+            null :
```
Comment (whiskeysierra) [inline server/checks/Approval.js]: Can that even happen? Does Github allow empty pull requests?

File: server/checks/Approval.js [hunk 2]
```diff
@@ -139,9 +138,12 @@ export default class Approval extends Check {
...
+            commits[commits.length - 1].committer.login
           const comments = await github.getComments(user, repo, number, formatDate(dbPR.last_push), token)
-          const countConfig = Object.assign({}, config.approvals, {ignore: [opener]})
+          const countConfig = Object.assign({}, config.approvals, {ignore: lastCommitter ? [lastCommitter] : []})
```
Comment (whiskeysierra) [inline server/checks/Approval.js]: I could imagine the different teams would want to configure it differently. I'd prefer to not count approvals from any committer. Others might prefer the current version of ignoring the pull request creator. Maybe ignore committer(s) and opener?

Comment (hjacobs) [inline server/checks/Approval.js]: @whiskeysierra I think only ignoring the PR opener is not good enough from the compliance POV (as the approver might sneak in stuff by committing right before he approves, i.e. no four eyes).

Comment (hjacobs) [inline server/checks/Approval.js]: @whiskeysierra but I like the idea that teams can decide to ignore approvals by opener AND last committer.

File: server/checks/Approval.js (hunk 1 above)
Comment (prayerslayer) [inline server/checks/Approval.js]: Probably not :grin:

Comment (jmcs) [inline server/checks/Approval.js]: I just tried and it's not, if the head of the compare branch is in the base branch, Github always marks the pull request as merged.

File: server/checks/Approval.js (hunk 2 above)
Comment (mfellner) [inline server/checks/Approval.js]: @whiskeysierra @hjacobs Opened a proposal for such an option here: #290

**Comment by mfellner** [comment]
👍

**Pull request merged**